File: Challenges/March Challenge 2020/ENGXOR.cpp
#include<iostream>
#include<bitset>
#include<algorithm>

using namespace std;

int getCount(unsigned int n){
    int result = 0;

    while(n > 0){
        if(n%2 == 1){
            result += 1;
        }
        n = n/2;
    }

    return result;
}

int main(){
    int t;
    ios_base::sync_with_stdio(false); 
    cin.tie(NULL);
    cin>>t;

    while (t--) {
        int N,Q;
        cin>>N>>Q;
        int questions[N];

        for(int i=0; i<N; i++){
            cin>>questions[i];
        }

        for(int i=0;i<Q;i++){
            int P;
            cin>>P;
            unsigned int evenCount=0;
            unsigned int oddCount = 0;

            for(int i=0;i<N;i++){
                unsigned int result = getCount(P ^ questions[i]);
                if(result%2 == 0)
                    evenCount += 1;
                else
                    oddCount += 1;
            }

            cout<<evenCount<<" "<<oddCount<<endl;

        }

    }
    
    return 0;
}

File: Challenges/March Challenge 2020/ENGXOR.py
import sys
import psyco

psyco.full()
t = int(sys.stdin.readline())
def getCount(n):
    result = 0
    while n > 0:
        if(n % 2 == 1):
            result += 1
        n = n//2
    return result
        
while t != 0:

    N, Q = map(int, sys.stdin.readline().split())
    questions = [int(x) for x in sys.stdin.readline().split()]
    for i in range(0,Q):
        P = int(input())
        evenCount = 0
        oddCount = 0
        
        for i in questions: 
            result = getCount(P ^ i)     
            if(result % 2 == 0):
                evenCount += 1
            else:
                oddCount += 1

        sys.stdout.write('{} {}\n'.format(evenCount, oddCount))

    t -= 1


